packages/playwright-test/src/reporters/github.ts

```diff
@@ -57,20 +60,52 @@ class GitHubLogger {
 }
 
 export class GitHubReporter extends BaseReporter {
-  githubLogger = new GitHubLogger();
+  private _githubLogger = new GitHubLogger();
+  private _options: Options = { annotations: 'on', summary: 'on' };
+
+  constructor(options?: Options) {
+    super();
+    this._options = { ...this._options, ...options };
+  }
 
   printsToStdio() {
     return false;
   }
 
   override async onEnd(result: FullResult) {
     super.onEnd(result);
-    this._printAnnotations();
+    if (this._options.annotations !== 'off')
+      this._printAnnotations();
+    if (this._options.summary !== 'off')
+      await this._writeGHASummary();
   }
 
   override onError(error: TestError) {
     const errorMessage = formatError(this.config, error, false).message;
-    this.githubLogger.error(errorMessage);
+    this._githubLogger.error(errorMessage);
+  }
+
+  private async _writeGHASummary() {
+    if (!GITHUB_STEP_SUMMARY_PATH)
+      return;
+
+    const allCases = this.suite.allTests();
+    const problematicCases = allCases.filter(t => PROBLEMATIC_OUTCOMES.includes(t.outcome()));
+    if (!problematicCases.length)
+      return;
+
+    const detailElements = sortByOutcomeAndTitle(problematicCases).map(t => [
+      `<details>`,
+      `  <summary>${summaryTitleElement(t)}</summary>`,
+      `  <pre>${escapeHTML(stripAnsiEscapes(formatFailure(this.config, t).message))}</pre>`,
+      `</details>`,
+    ].join('\n'));
+
+    await fs.promises.appendFile(GITHUB_STEP_SUMMARY_PATH, [
+      summaryBadgeElement(allCases),
+      ...detailElements,
+      '',
+    ].join('\n')).catch(e => this._githubLogger.warning(`Playwright Test GitHub Reporter failed to write GitHub Summary: ${e}`));
```

Comment:
Move this catch up onto the callsite instead: https://github.com/microsoft/playwright/pull/14763/files#diff-22c9962e5f77e21b1ab9611b86c2ca7512794d31dced527e6680d275a596bd1aR80